src/nerpa_pipeline/predictions_preprocessor.py

```diff
@@ -9,33 +9,33 @@
 import handle_E
 import splitter
 import handle_helper
-
-def gen_prediction_dict(orf_part, input_file_name, dirname):
-    double_orf, double_aa = handle_PCP2.get_double_orfs_and_AA(dirname, orf_part)
-    mt_aa = handle_MT.get_MT_AA(dirname, orf_part)
-    d_aa = handle_E.get_D_AA(dirname, orf_part)
-
-    prediction_dict = {}
+from collections import defaultdict
+
+def gen_prediction_dict(orf_modules_names, input_file_name, dirname):
+    '''
+    dirname: converted_antiSMASH_v5_outputs
+    input_file_name: CONTIGNAME_nrpspredictor2_codes.txt
+    orf_part: list of the form ['ctg1_trsI_A1', 'ctg1_trsI_A2']
+    '''
+    double_orf, double_aa = handle_PCP2.get_double_orfs_and_AA(dirname, orf_modules_names)  # the lists of orfs and modules which can be duplicated
+    mt_aa = handle_MT.get_MT_AA(dirname, orf_modules_names)  # the list of modules with methylation
+    d_aa = handle_E.get_D_AA(dirname, orf_modules_names)  # the list of modules with epimerization
+
+    prediction_dict = defaultdict(lambda: '')
     with open(input_file_name, 'r') as rf:
         for line in rf:
-            ctgorf = '_'.join(line.split('_')[:2])
+            module_name, aa_prediction, scores = line.split('\t')
+            ctgorf, domain_id = module_name.rsplit('_', 1)
```

Comment:
If I understand correctly, `ctgorf` might have multiple '_' inside, e.g., 'ctg123_erf_dba'. Could it be a problem in the downstream processing? The C++ part might expect exactly one '_' separating ORF name and Module name..

Reply:
I fixed the C++ part so that it can handle multiple _